src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -3284,10 +3538,7 @@ public int parse(DateTimeParseContext context, CharSequence text, int position)
          * @throws DateTimeException if the value cannot be converted to a fraction
          */
         private BigDecimal convertToFraction(long value) {
-            ValueRange range = field.range();
-            range.checkValidValue(value, field);
-            BigDecimal minBD = BigDecimal.valueOf(range.getMinimum());
-            BigDecimal rangeBD = BigDecimal.valueOf(range.getMaximum()).subtract(minBD).add(BigDecimal.ONE);
+            field.range().checkValidValue(value, field);
             BigDecimal valueBD = BigDecimal.valueOf(value).subtract(minBD);
             BigDecimal fraction = valueBD.divide(rangeBD, 9, RoundingMode.FLOOR);
             // stripTrailingZeros bug
```

Comment:
I believe this bug was fixed a while back, so this code could be simplified.

Reply:
Got a reference to which bug this was and how it manifests?

Reply:
If you're referring to JDK-6480539: "BigDecimal.stripTrailingZeros() has no effect on zero itself ("0.0")", it was fixed in JDK 8.